File: 2024-maximize-the-confusion-of-an-exam/2024-maximize-the-confusion-of-an-exam.py
class Solution:
    def maxConsecutiveAnswers(self, answerKey: str, k: int) -> int:
        answer = 0
        maxcount = 0
        table = defaultdict(int)
        start = 0
        for idx, a in enumerate(answerKey):
            table[a == "T"] += 1
            maxcount = max(maxcount, table[a == "T"])
            while maxcount + k < idx - start + 1:
                table[answerKey[start] == "T"] -= 1
                start += 1
            answer = max(answer, idx - start + 1)
        return answer

File: leetcode/Medium/2024-maximize-the-confusion-of-an-exam/2024-maximize-the-confusion-of-an-exam.c
int maxConsecutiveAnswers(char *answerKey, int k) {
  int answer = 0;
  int maxcount = 0;
  int t = 0;
  int f = 0;
  int start = 0;
  int end = 0;
  while (answerKey[end] != NULL) {
    if (answerKey[end] == 'T') {
      t++;
      if (maxcount < t)
        maxcount = t;
    } else {
      f++;
      if (maxcount < f)
        maxcount = f;
    }

    while (maxcount + k < end - start + 1) {
      if (answerKey[start] == 'T') {
        t--;
      } else {
        f--;
      }
      start++;
    }

    if (answer < end - start + 1)
      answer = end - start + 1;

    end += sizeof(char);
  }
  return answer;
}